abi-versions/vNEXT/README.md

```diff
@@ -0,0 +1,1471 @@
+# Proxy-Wasm vNEXT ABI specification
+
+# Callbacks and Functions
+
+### Callbacks optionally exposed by the Wasm module
+
+All callbacks (entry points named `proxy_on_<event>`) are optional, and will be
+called only if exposed by the Wasm module.
+
+All callbacks, other than the integration functions, include unique context
+identifier as the first parameter, which can be used to distinguish between
+different contexts.
+
+### Functions exposed by the host environment
+
+All functions exposed by the host environment return `proxy_result_t`, which
+indicates the status of the call (success, invalid memory access, etc.), and
+return values are written into memory pointed by `return_<value>` parameters.
+
+
+## Integration
+
+### Functions exposed by the host environment
+
+#### `proxy_abi_version_X_Y_Z`
+
+* params:
+  - none
+* returns:
+  - none
+
+Exports ABI version (vX.Y.Z) that the Wasm module was compiled against.
+
+#### `proxy_log`
```

Comment:
Can we also implement "proxy_get_log_level?" It's handy to initialize the logging library with the caveat that one shouldn't call it on every long message and that we haven't figured out how to call back when the log level changes.

Reply:
Yeah, I'll add both.